pkg/controller/clusterset/node.go

```diff
@@ -43,33 +46,36 @@ func AddNodeController(ctx context.Context, mgr manager.Manager) error {
 }
 
 func (n *NodeReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := n.logger.With("Node", req.NamespacedName)
 	var clusterSetList v1alpha1.ClusterSetList
 	if err := n.Client.List(ctx, &clusterSetList); err != nil {
-		return reconcile.Result{}, k3kcontroller.LogAndReturnErr("failed to list clusterSets", err)
+		return reconcile.Result{}, err
 	}
 
 	if len(clusterSetList.Items) <= 0 {
 		return reconcile.Result{}, nil
 	}
 
-	if err := n.ensureNetworkPolicies(ctx, clusterSetList); err != nil {
+	if err := n.ensureNetworkPolicies(ctx, clusterSetList, log); err != nil {
 		return reconcile.Result{}, err
 	}
 
 	return reconcile.Result{}, nil
 }
 
-func (n *NodeReconciler) ensureNetworkPolicies(ctx context.Context, clusterSetList v1alpha1.ClusterSetList) error {
+func (n *NodeReconciler) ensureNetworkPolicies(ctx context.Context, clusterSetList v1alpha1.ClusterSetList, log *zap.SugaredLogger) error {
```

Comment:
Does it need to be passed in as an arg since it's on the struct?

Reply:
short answer is yes, the log is indeed in the structure however, for each reconcile loop the log object is initiated with a .With fields to describe the object name in that reconcile loop and this one gets passed around through the functions, its more lines of code but its the convention with logging in controller-runtime apparently